### PR TITLE
Added `uf2conv` tool

### PR DESCRIPTION
**Pull request by HU90m**
A simple python script I'm using to create uf2 files from binaries. This will be needed for the sonata hackathon.

### CODE REVIEW

**Comment by HU90m** [comment]
`elf2uf2-rs` is taylored to the pi pico, so wasn't immediately useful, e.g.

**Review by marnovandermaas — APPROVED** [review]
Sounds fine to me. No harm adding it to the nix environment.

**Review by hcallahan-lowrisc — APPROVED** [review]

**Comment by HU90m** [comment]
Thanks for the quick reviews!

File: pkgs/uf2conv.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  src = fetchFromGitHub {
+    owner = "makerdiary";
+    repo = "uf2utils";
+    rev = "61d9f3ff4913461b9813e666cce62b022652144a";
+    hash = "sha256-3mXOvzkHmTV0k8qoy2k5TGBFX2l9EgPuaZPLyTdBHzQ=";
+  };
+  meta = {
+    description = "uf2conv is an open source Python based tool for packing and unpacking UF2 files.";
+    homepage = "https://github.com/makerdiary/uf2utils";
+    license = lib.licenses.mit;
```
Comment (nbdd0121) [inline pkgs/uf2conv.nix:21]: This is missing `mainProgram` causing `nix run` to fail.

File: pkgs/uf2conv.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright lowRISC contributors.
+#
+# SPDX-License-Identifier: MIT
+{
+  python3Packages,
+  fetchFromGitHub,
+  lib,
+}:
+python3Packages.buildPythonPackage rec {
+  name = "uf2conv";
```
Comment (hcallahan-lowrisc) [inline pkgs/uf2conv.nix:10]: This should be `pname` actually. Nix run can then guess the binary name correctly too.

**Pull request merged**